### DISCUSSION TITLE
Media Query based ViewTransitions

### DISCUSSION THREAD

**Discussion by Firaenix** [withastro/roadmap#838] [Proposal]
# Summary
This RFC proposes the addition of a feature that allows users to specify ViewTransitions based on Media Query, enabling different view transitions for different devices or screen sizes.

# Background & Motivation
Currently, the system does not provide a way to customize view transitions based on media queries as far as I know.

For example:
- I want to use the Slide transition on Mobile but Fade on Desktop
- I want to disable transitions on Desktop but other screen sizes are unaffected

# Goals
	- Enable the specification of ViewTransitions based on Media Queries.
	- Provide flexibility in defining different view transitions for different devices or screen sizes.
	- Improve the user experience by allowing customized view transitions based on specific device requirements.

**Comment by florian-lefebvre** [comment]
That sounds interesting! What do you think @martrapp ? Could that be an idea for vtbot if it can't make it to core?

**Comment by martrapp** [comment]
@Firaenix at least the simpler examples should be possible right now via CSS

Switch off animation for landscape mode
```
@media (orientation: landscape) {
  ::view-transition-old(*),
  ::view-transition-new(*) {
    animation: none;
  }
}
```

Likewise you should be able to use media queries to override the keyframes definition for the animation being used in the view transition.

On large screens, replace slide with an rotation effect
```
@media screen and (min-width: 1400px) {
  @keyframes astroSlideFromRight {
    from {
      transform: rotateZ(0);
    }
    to {
      transform: rotateZ(180deg);
    }
  }
  ...
}
```
As @florian-lefebvre suggested, I would happily add some examples/howtos to [astro-vtbot](https://events-3bg.pages.dev/). 

As for media query-based view transitions in the Astro core: that depends very much on how simple and maintainable such an addition could be. Do you already have ideas on how you envision this?
